lib/executor/ramping_arrival_rate.go

```diff
@@ -263,7 +263,7 @@ func (varc RampingArrivalRateConfig) cal(et *lib.ExecutionTuple, ch chan<- time.
 			}
 		} else {
 			endCount += dur * to
-			for ; i <= endCount; i += float64(next()) {
+			for ; i <= math.Ceil(endCount); i += float64(next()) {
```

Comment:
I feel like this should be the fix. 

The reported bug is mostly a visual one, where we tell an executor did not finish, but it ... did.

This is confirmed by the test time below changing, which should not happen. 

Maybe I am missing some very small detail which makes the below times previously wrong. But the current numbers seem to progress with the correct steps between them while with your change you move them around a bit.

IMO the fix should be in the thing reporting to the progressbar or the progressbar itself to not think of this as an interruption